src/test/java/org/kiwiproject/beta/slf4j/TimestampingLoggerTest.java

```diff
@@ -222,12 +228,28 @@ private void assertAppendedElapsedEventMessages(Level expectedLevel) {
                         .endsWith(" millis]"));
     }
 
+    @AllArgsConstructor
+    static class LoggedMessages {
+        final List<String> logMessages;
+        final List<String> elapsedTimeMessages;
+
+        /**
+         * Split all event messages into two lists: one containing the log messages, the other containing elapsed time
+         * messages. The input list must have an even number of elements.
+         */
+        static LoggedMessages from(List<String> eventMessages) {
+            checkEvenItemCount(eventMessages);
+            Map<Boolean, List<String>> result = eventMessages.stream()
+                    .collect(partitioningBy(str -> eventMessages.indexOf(str) % 2 == 0));
```

Comment:
looking up the index of the steamed item in the source collection feels icky, maybe an old school for-loop or and IntStream?

Reply:
Yes. This is the code that ChatGPT suggested! If this were in production code I would definitely find another way, but figured it was probably fine in a test with a very small collection. Any suggestions? Maybe you can talk to ChatGPT and get it to produce better code. When I asked it to rewrite it using the [StreamEx](https://github.com/amaembo/streamex) library, that's when it went off the rails and started telling me to use methods that _do not exist_ in the `StreamEx` class. So I stuck with this original suggestion, which again I admit is sub-par.

Reply:
Here it is as an old school for loop:

```java
List<String> logMessages = new ArrayList<>();
List<String> elapsedTimeMessages = new ArrayList<>();
for (int i = 0; i < eventMessages.size(); i++) {
    var message = eventMessages.get(i);
    if (i % 2 == 0) {
        logMessages.add(message);
    } else {
        elapsedTimeMessages.add(message);
    }
}
```

Reply:
I "asked" IntelliJ if it could convert this into the Stream API and it politely declined. It offered to replace with a `while` or forEach (using `IntStream.range.forEach`) or forEachOrdered (using `IntStream.range.forEachOrdered`), none of which are any better. I think the above is too "complicated" to simplify into the Stream API because of the two lists.

Reply:
Ha, I fed the `for` loop code to ChatGPT and asked it to convert to the Stream API. Here's what it gave me:

```java
var logMessages = IntStream.range(0, eventMessages.size())
    .filter(i -> i % 2 == 0)
    .mapToObj(eventMessages::get)
    .collect(Collectors.toList());

var elapsedTimeMessages = IntStream.range(0, eventMessages.size())
    .filter(i -> i % 2 == 1)
    .mapToObj(eventMessages::get)
    .collect(Collectors.toList());
```

So, while this looks correct (I did not try it), it streams the collection twice, which is not ideal.

Reply:
And when I asked it to change the code so it only streams the collection one time, it came up with:

```java
Map<Boolean, List<String>> result = IntStream.iterate(0, i -> i < eventMessages.size(), i -> i + 1)
    .mapToObj(eventMessages::get)
    .collect(Collectors.partitioningBy(i -> eventMessages.indexOf(i) % 2 == 0));

List<String> logMessages = result.get(true);
List<String> elapsedTimeMessages = result.get(false);
```

I did try this and it works, though I don't really like it. It is not at all an idiomatic usage of the Stream API.

So, I guess for loop it shall be!